### PR TITLE
Orders entity

### PR DESCRIPTION
**Pull request by Vitaliy-Maznyi**

### CODE REVIEW

**Comment by Vitaliy-Maznyi** [comment]
@yashka713 @DenisUA @PeterKujda please comment and approve :suspect:

File: app/models/order.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Order < ApplicationRecord
+  belongs_to :user
+  belongs_to :apartment
+
+  enum status: { booked: 0, in_progress: 1, cancelled: 2, expired: 3 }
+
+  validates :total, :start_from, :end_at, :user_id, :apartment_id, presence: true
+  validates :total, numericality: { greater_than: 0 }
```
Comment (yashka713) [inline app/models/order.rb:8]: 7 & 8 looks the same, do they need look the same? I am speaking about `:total`

File: app/views/orders/edit.slim [hunk 2]
```diff
@@ -0,0 +1,3 @@
+| edit
+
+= render 'orders/form'
```
Comment (yashka713) [inline app/views/orders/edit.slim:3]: Please, add the endline

File: app/models/order.rb (hunk 1 above)
Comment (Vitaliy-Maznyi) [reply app/models/order.rb:8]: on the 7 line is enumeration of fields which has to be presensed
on the 8 line only fields which have to be more than 0 so it's not the same

File: app/views/orders/edit.slim (hunk 2 above)
Comment (Vitaliy-Maznyi) [reply app/views/orders/edit.slim:3]: it's already here

**Comment by yashka713** [comment]
:1st_place_medal:

**Review by PetroKuida — APPROVED** [review]
seems ok.

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -2,3 +2,10 @@
 //= require jquery
 //= require jquery_ujs
 //= require bootstrap-sprockets
+//= require bootstrap-datepicker
+
+$(document).ready(function(){
```
Comment (DenisUA) [inline app/assets/javascripts/application.js]: This doesn't belong to the application.js. Please create `orders.js` file and put this code inside

File: app/assets/javascripts/application.js [hunk 4]
```diff
@@ -2,3 +2,10 @@
...
+  $('#new_order .input-daterange').datepicker({
```
Comment (DenisUA) [inline app/assets/javascripts/application.js]: Please add prefix `js` to all classes that you are using inside js

File: app/controllers/orders_controller.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
+class OrdersController < ApplicationController
+  before_action :authenticate_user!, except: %i[index show]
```
Comment (DenisUA) [inline app/controllers/orders_controller.rb]: Please move it ApplicationController

File: app/controllers/orders_controller.rb [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  before_action :set_order, only: %i[show edit update destroy]
+
+  def index
+    @orders = Order.all
+  end
+
+  def new
+    @order = Order.new
+    @apartments = Apartment.where(status: 0)
```
Comment (DenisUA) [inline app/controllers/orders_controller.rb:10]: This line shouldn't be here

File: app/controllers/orders_controller.rb [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  end
+
+  def edit
+    respond_to do |format|
+      if @order.cancelled?
```
Comment (DenisUA) [inline app/controllers/orders_controller.rb]: As Admin I would like to change any fields. It's not possible with this code.

File: app/controllers/orders_controller.rb [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        format.json { render json: @order }
+      else
+        format.html { render :edit }
+        format.json { render json: @order.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @order.cancelled!
```
Comment (DenisUA) [inline app/controllers/orders_controller.rb:50]: What if any error happens?

File: app/models/order.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+class Order < ApplicationRecord
+  belongs_to :user
+  belongs_to :apartment
+
+  enum status: { booked: 0, in_progress: 1, cancelled: 2, expired: 3 }
+
+  validates :total, :start_from, :end_at, :user_id, :apartment_id, presence: true
```
Comment (DenisUA) [inline app/models/order.rb:7]: Please add validation that start_from can't be in past

File: app/views/orders/edit.slim (hunk 2 above)
Comment (DenisUA) [reply app/views/orders/edit.slim:3]: nope, it's missing.

File: app/views/orders/index.slim [hunk 10]
```diff
@@ -0,0 +1,25 @@
+= link_to 'Create new Order', new_order_path, class: 'btn btn-success'
+.row
+  .col-lg-12
+    table.table.table-hover
+      thead
+        tr
+          th= 'id'
```
Comment (DenisUA) [inline app/views/orders/index.slim:7]: You don't need extra interpolation. 
`th id`
From here to line 13

File: app/views/orders/show.slim [hunk 11]
```diff
@@ -0,0 +1,20 @@
+h3 Show Page
+p
+  | User: #{ @order.user.full_name }
```
Comment (DenisUA) [inline app/views/orders/show.slim]: Can be simplified to:
```
' User:
= @order.user.full_name
```

**Review by DenisUA — CHANGES_REQUESTED** [review]

File: app/views/orders/edit.slim (hunk 2 above)
Comment (Vitaliy-Maznyi) [reply app/views/orders/edit.slim:3]: https://github.com/DenisUA/Ruby-hotel/pull/15/commits/3a72381d9240a6987a893cae6a06fcf10c56f0cd 
I've already fix this

File: app/controllers/orders_controller.rb (hunk 6 above)
Comment (Vitaliy-Maznyi) [reply app/controllers/orders_controller.rb:10]: Is it better to use such way?
```
= f.input :apartment_id, label: 'Apartment',
          collection: Appartment.available,
          label_method: -> (collection){ "room number: #{collection.room_number}" },

```

File: app/views/orders/index.slim (hunk 10 above)
Comment (Vitaliy-Maznyi) [reply app/views/orders/index.slim:7]: I can't remove `' '` because it won't be shown as string

File: app/controllers/orders_controller.rb (hunk 8 above)
Comment (Vitaliy-Maznyi) [reply app/controllers/orders_controller.rb:50]: done

File: app/models/order.rb (hunk 9 above)
Comment (Vitaliy-Maznyi) [reply app/models/order.rb:7]: done

File: app/controllers/orders_controller.rb (hunk 6 above)
Comment (Vitaliy-Maznyi) [reply app/controllers/orders_controller.rb:10]: done https://github.com/DenisUA/Ruby-hotel/pull/15/commits/c4ce893a5e1b82b485aa3b6b3dc490aaa3560152

File: app/views/orders/index.slim (hunk 10 above)
Comment (Vitaliy-Maznyi) [reply app/views/orders/index.slim:7]: done https://github.com/DenisUA/Ruby-hotel/commit/668f633656029069392d287921150d46d855cfab

**Pull request merged**